src/firefly/js/charts/ui/PlotlyWrapper.jsx

```diff
@@ -25,6 +27,21 @@ const defaultConfig= {
     ]
 };
 
+export function downloadChart(chartId) {
+    getPlotLy().then( (Plotly) => {
+        const chartDiv = document.getElementById(chartId);
+        if (chartId && chartDiv) {
+            downloadCounter++;
+            Plotly.downloadImage(chartDiv, {
+                format: 'png',
+                filename: `plotly-${chartId.replace(/\-.*$/, downloadCounter)}`
```

Comment:
I think 'replace' and counter tracking is unnecessary.  Even 'plotly', I don't think it's needed.
Filename should be chart.title or chart.chartId if title not available.